src/datatypes2/src/value.rs

```diff
@@ -126,9 +126,9 @@ impl Value {
             Value::Float64(_) => ConcreteDataType::float64_datatype(),
             Value::String(_) => ConcreteDataType::string_datatype(),
             Value::Binary(_) => ConcreteDataType::binary_datatype(),
-            // Value::List(list) => ConcreteDataType::list_datatype(list.datatype().clone()),
+            Value::List(list) => ConcreteDataType::list_datatype(list.datatype().clone()),
             Value::Date(_) => ConcreteDataType::date_datatype(),
-            // Value::DateTime(_) => ConcreteDataType::datetime_datatype(),
+            Value::DateTime(_) => ConcreteDataType::datetime_datatype(),
```

Comment:
Why it occurs in this patch

Reply:
I just found out that we forgot to uncomment this when implementing `DateTimeType`. Do I need open a new PR to do this?

Reply:
no need